### PR TITLE
use ldply instead of lapply to tidy aovlist's

### PR DESCRIPTION
**Pull request by mvevans89**
this changes the way tidy is used to tidy aovlist's (such as within-subject error structures) to close issue #299

### CODE REVIEW

File: R/anova_tidiers.R [hunk 1]
```diff
@@ -103,22 +103,28 @@ tidy.aov <- function(x, ...) {
...
   }
 
-  # ret <- plyr::ldply(x, tidy, .id = "stratum")
-  ret <- lapply(x, function(a) tidy(stats::anova(a)))
-  ret <- lapply(
-    names(ret),
-    function(a) dplyr::mutate(ret[[a]], stratum = a)
-  )
-  ret <- do.call("rbind", ret)
+  ret <- plyr::ldply(x, tidy, .id = "stratum")
```
Comment (alexpghayes) [inline R/anova_tidiers.R]: We're looking to move away from `plyr` at the moment. Would you be willing to use `purrr` here instead?

File: R/anova_tidiers.R [hunk 2]
```diff
@@ -103,22 +103,28 @@ tidy.aov <- function(x, ...) {
 
 
 #' @rdname anova_tidiers
+#' 
+#' @importFrom plyr ldply
+#' 
+#' @import dplyr
```
Comment (alexpghayes) [inline R/anova_tidiers.R]: `dplyr` and `plyr` get imported package-wide in DESCRIPTION, so I believe these aren't necessary.

File: tests/testthat/test-anova.R [hunk 3]
```diff
@@ -0,0 +1,15 @@
+context("anova tidiers")
```
Comment (alexpghayes) [inline tests/testthat/test-anova.R]: Regression tests! Awesome!

File: tests/testthat/test-anova.R [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+test_that("tidy.aovlist works"){
+  aovlistfit <- aov(mpg ~ wt + disp + Error(drat), mtcars)
+  td <- suppressWarnings(tidy(aovlistfit))
```
Comment (alexpghayes) [inline tests/testthat/test-anova.R]: What kind of warnings are getting suppressed here?

**Comment by alexpghayes** [comment]
Don't worry about the build failure at the moment, BTW, that's my fault and I need to fix it.

File: R/anova_tidiers.R (hunk 2 above)
Comment (mvevans89) [reply R/anova_tidiers.R]: Got it. I'll remove them.

File: tests/testthat/test-anova.R (hunk 4 above)
Comment (mvevans89) [reply tests/testthat/test-anova.R]: I checked and there actually are none. Warning suppression was just a hold over from copying the format of the `test-lm.R`. I'll take it out

File: R/anova_tidiers.R (hunk 1 above)
Comment (mvevans89) [reply R/anova_tidiers.R]: Works for me. I'll switch it up.

**Comment by alexpghayes** [comment]
Changes look good. Will merge as soon as I sort out the `lavaan` stuff causing the build failure.

**Pull request merged**